fast64_internal/oot/oot_upgrade.py

```diff
@@ -313,16 +315,23 @@ def upgradeActors(actorObj: Object):
                     break
     elif actorObj.ootEmptyType == "Transition Actor":
         transActorProp = actorObj.ootTransitionActorProperty
-        transActorProp.isRoomTransition = actorObj["ootTransitionActorProperty"]["dontTransition"] == False
-        del actorObj["ootTransitionActorProperty"]["dontTransition"]
-
-        if transActorProp.isRoomTransition:
+        if "dontTransition" in transActorProp or "roomIndex" in transActorProp:
             for obj in bpy.data.objects:
-                if obj.type == "EMPTY":
-                    if obj.ootEmptyType == "Room":
-                        if actorObj in obj.children_recursive:
-                            transActorProp.fromRoom = obj
-
-                        if obj.ootRoomHeader.roomIndex == actorObj["ootTransitionActorProperty"]["roomIndex"]:
-                            transActorProp.toRoom = obj
-                            del actorObj["ootTransitionActorProperty"]["roomIndex"]
+                if obj.type == "EMPTY" and obj.ootEmptyType == "Room" and actorObj in obj.children_recursive:
+                    transActorProp.fromRoom = obj
+                    break
+
+            if "dontTransition" in transActorProp:
+                transActorProp.isRoomTransition = transActorProp["dontTransition"] == False
+                del transActorProp["dontTransition"]
+            elif transActorProp.fromRoom is not None:
```

Comment:
why is the second block under an else now? (elif)

Reply:
bump

Reply:
this case is most likely impossible but I thought the for may not find an object to set to ``transActorProp.fromRoom``, so if this is still ``None`` on the block you highlighted it will raise an error (I think I should just assert this idk)

Reply:
I mean `if "dontTransition" in transActorProp:` is likely to pass if upgrading a blend
then that means `transActorProp.toRoom` won't be set based on upgrading from `transActorProp["roomIndex"]`